service/controller/v22/resource/asgstatus/create.go

```diff
@@ -0,0 +1,106 @@
+	"fmt"
+
+	"github.com/aws/aws-sdk-go/service/autoscaling"
+	"github.com/giantswarm/microerror"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/giantswarm/aws-operator/service/controller/v22/controllercontext"
+	"github.com/giantswarm/aws-operator/service/controller/v22/key"
+)
+
+func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
+	cr, err := key.ToCustomObject(obj)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+	cc, err := controllercontext.FromContext(ctx)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	workerASGName := cc.Status.Drainer.WorkerASGName
+	if workerASGName == "" {
+		r.logger.LogCtx(ctx, "level", "debug", "message", "worker ASG name is not available yet")
+		r.logger.LogCtx(ctx, "level", "debug", "message", "canceling resource")
+		return nil
+	}
+
+	var asg *autoscaling.Group
+	{
+		r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("finding ASG %#q", workerASGName))
+
+		i := &autoscaling.DescribeAutoScalingGroupsInput{
+			AutoScalingGroupNames: []*string{
+				&workerASGName,
+			},
+		}
+		o, err := cc.AWSClient.AutoScaling.DescribeAutoScalingGroups(i)
+		if err != nil {
+			return microerror.Mask(err)
+		}
+
+		if len(o.AutoScalingGroups) != 1 {
+			return microerror.Maskf(executionFailedError, "there must be one item for ASG %#q", workerASGName)
+		}
+		asg = o.AutoScalingGroups[0]
+
+		r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("found ASG %#q", workerASGName))
+	}
+
+	var desiredCapacity int
+	{
+		if asg.DesiredCapacity == nil {
+			return microerror.Maskf(executionFailedError, "desired capacity must not be empty for ASG %#q", workerASGName)
+		}
+		desiredCapacity = int(*asg.DesiredCapacity)
+		r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("desired capacity of %#q is %d", workerASGName, desiredCapacity))
+	}
+
+	var maxSize int
+	{
+		if asg.MaxSize == nil {
+			return microerror.Maskf(executionFailedError, "max size must not be empty for ASG %#q", workerASGName)
+		}
+		maxSize = int(*asg.MaxSize)
+		r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("max size of %#q is %d", workerASGName, maxSize))
+	}
+
+	var minSize int
+	{
+		if asg.MinSize == nil {
+			return microerror.Maskf(executionFailedError, "min size must not be empty for ASG %#q", workerASGName)
+		}
+		minSize = int(*asg.MinSize)
+		r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("min size of %#q is %d", workerASGName, minSize))
+	}
+
+	{
+		r.logger.LogCtx(ctx, "level", "debug", "message", "updating status with desired capacity")
+
+		newObj, err := r.g8sClient.ProviderV1alpha1().AWSConfigs(cr.GetNamespace()).Get(cr.GetName(), metav1.GetOptions{})
+		if err != nil {
+			return microerror.Mask(err)
+		}
+
+		newObj.Status.Cluster.Scaling.DesiredCapacity = desiredCapacity
+
+		_, err = r.g8sClient.ProviderV1alpha1().AWSConfigs(newObj.GetNamespace()).UpdateStatus(newObj)
+		if err != nil {
+			return microerror.Mask(err)
+		}
```

Comment:
Should the reconciliation be cancelled here?

If we cancel reconciliation here there is no need to set that information in the context. It can be taken from `cr.Status` in subsequent loops.

Reply:
See https://github.com/giantswarm/aws-operator/pull/1397. Cancelation makes sense. I did it in the PR. The status and context thing is a bit off, agreed. The thing is that we only have the desired capacity in the status but also need the min and max size in the `cloudformation` resource later on. We are planning some refactoring for the status thingy also because of the CF stack transition weirdness blocking certain updates.

Reply:
Maybe it makes sense to extend the status then?

Reply:
We will definitely consider this in the future. You are right that both is weird. We agreed on going with what we have though because AS has to be released next week. Thank you for your feedback here though. Emphasized the right thoughts.